src/main/java/de/maxhenkel/pipez/blocks/tileentity/render/PipeRenderer.java

```diff
@@ -17,16 +17,16 @@
 import net.minecraft.util.ResourceLocation;
 import net.minecraft.util.math.vector.Quaternion;
 import net.minecraft.util.math.vector.Vector3f;
-import net.minecraftforge.client.event.ModelBakeEvent;
 import net.minecraftforge.client.model.ModelLoader;
 import net.minecraftforge.client.model.data.EmptyModelData;
-import net.minecraftforge.eventbus.api.SubscribeEvent;
-import net.minecraftforge.fml.javafmlmod.FMLJavaModLoadingContext;
 
+import java.util.ArrayList;
 import java.util.List;
 
 public abstract class PipeRenderer extends TileEntityRenderer<PipeTileEntity> {
 
+    private static List<PipeRenderer> renderers = new ArrayList();
```

Comment:
There has to be a cleaner method than storing all renderers in a list or registering a event for every renderer.

Reply:
Every renderer has a cached model, and all cached models have to be invalidated. How about registering all cached models in ModelEvents?

Reply:
This might be a better solution.